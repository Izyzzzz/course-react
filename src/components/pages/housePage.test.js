import React from 'react';
import HousePage from './housePage';
import {shallow} from 'enzyme';

describe('Testing <HousePage/>', () => {
    const house = shallow(<HousePage/>);
    describe('Testing snap & state', () => {
        it('HousePage have rendered correctly', () => {        
            expect(house).toMatchSnapshot();
        });
        it('HousePage state "error" is false', () => {
            expect(house.state().error).toBeFalsy();
        });
    });
});