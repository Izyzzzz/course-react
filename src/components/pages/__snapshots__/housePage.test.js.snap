// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <HousePage/> Testing snap & state HousePage have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HousePage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "left": <ItemList
        getData={[Function]}
        onItemSelected={[Function]}
        renderItem={[Function]}
      />,
      "right": <ItemDetails
        getDataId={[Function]}
        plaseSelect="Please select a house"
      >
        <Field
          field="region"
          label="Region"
        />
        <Field
          field="words"
          label="Words"
        />
        <Field
          field="titles"
          label="Titles"
        />
        <Field
          field="overload"
          label="Overload"
        />
        <Field
          field="ancestralWeapons"
          label="Ancestral Weapons"
        />
      </ItemDetails>,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "left": <ItemList
          getData={[Function]}
          onItemSelected={[Function]}
          renderItem={[Function]}
        />,
        "right": <ItemDetails
          getDataId={[Function]}
          plaseSelect="Please select a house"
        >
          <Field
            field="region"
            label="Region"
          />
          <Field
            field="words"
            label="Words"
          />
          <Field
            field="titles"
            label="Titles"
          />
          <Field
            field="overload"
            label="Overload"
          />
          <Field
            field="ancestralWeapons"
            label="Ancestral Weapons"
          />
        </ItemDetails>,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
